darwin/client.py

```diff
@@ -197,9 +203,9 @@ def delete(
         endpoint: str,
         team: Optional[str] = None,
         retry: bool = False,
-        error_handlers: Optional[list] = None,
+        error_handlers: Optional[List[Callable[[int, str], None]]] = None,
```

Comment:
`Callable[[int, str], None]` to me is a good candidate to be made into its own type
```python
ErrorHandler = Callable[[int, str], None]
```

Also, can't we make `error_handler` non-optional, defaulting to `[]`? Not sure if that's going to be a problem tbh